apollo/interfaces/lambda_function/aws_utils.py

```diff
@@ -0,0 +1,18 @@
+from botocore.config import Config
+
+
+def get_retrieve_current_image_boto_config(connection_timeout: int) -> Config:
```

Comment:
nit: While the initial use case for this function is for boto client config to retrieve the current image URI from CloudFormation, it could easily serve other use cases. Maybe consider a more generic name (e.g. `get_boto_config`), and allowing `max_attempts` to be passed in (with a default=4).